plugin/timeouts/README.md

```diff
@@ -0,0 +1,58 @@
+# timeouts
+
+## Name
+
+*timeouts* - allows you to configure the server read, write and idle timeouts for the TCP, TLS and DoH servers.
+
+## Description
+
+CoreDNS is configured with sensible timeouts for server connections by default. However in some cases
+for example where CoreDNS is serving over a slow mobile data connection the default timeouts are not
+optimal.
+
+Additionally some routers hold open connections when using DNS over TLS or DNS over HTTPS. Allowing
+a longer idle timeout helps performance and reduces issues with such routers.
+
+The *timeouts* "plugin" allows you to configure CoreDNS server read, write and idle timeouts.
+
+## Syntax
+
+~~~ txt
+timeouts READ [WRITE] [IDLE]
+~~~
+
+For any timeouts that are not provided, default values are used which vary depending on the server type.
+
+## Examples
+
+Start a DNS-over-TLS server that picks up incoming DNS-over-TLS queries on port 5553 and uses the
+nameservers defined in `/etc/resolv.conf` to resolve the query. This proxy path uses plain old DNS.
+A 10 second read timeout, 20 second write timeout and a 60 second idle timeout have been configured.
+
+~~~
+tls://.:5553 {
+	tls cert.pem key.pem ca.pem
+	timeouts 10 20 60
```

Comment:
also I would expect these arguments to accept format in GO duration, meaning  `5s`, `500ms`, etc

Reply:
The revision I am going to submit will accept Go duration but if a plain number is specified it will assume that seconds was meant.